src/graphnet/models/graphs/nodes/nodes.py

```diff
@@ -211,3 +212,125 @@ def _construct_nodes(self, x: torch.Tensor) -> Data:
             raise AttributeError
 
         return Data(x=torch.tensor(array))
+
+
+class IceMixNodes(NodeDefinition):
+    """Calculate ice properties and perform random sampling.
+
+    Ice properties are calculated based on the z-coordinate of the pulse. For
+    each event, a random sampling is performed to keep the number of pulses
+    below a maximum number of pulses if n_pulses is over the limit.
+    """
+
+    def __init__(
+        self,
+        input_feature_names: Optional[List[str]] = None,
+        max_pulses: int = 768,
+        z_name: str = "dom_z",
+        hlc_name: str = "hlc",
+    ) -> None:
+        """Construct `IceMixNodes`.
+
+        Args:
+            input_feature_names: Column names for input features. Minimum
+            required features are z coordinate and hlc column names.
+            max_pulses: Maximum number of pulses to keep in the event.
+            z_name: Name of the z-coordinate column.
+            hlc_name: Name of the `Hard Local Coincidence Check` column.
+        """
+        super().__init__(input_feature_names=input_feature_names)
+
+        if input_feature_names is None:
```

Comment:
You are hardcoding these variables because you need to construct a feature index in the `__init__` method that you can later search through  in `_construct_nodes`, right?

If so, I think this is a great hint that we should look into changing the interplay between `GraphDefinition` and `NodeDefinition` so this information is available by default. Because as your code is written, it will fail if this model is run on a different detector class without passing the input features explicitly. I think it's fair to not change this in this PR as it's already quite large, but we should follow this up soon.